video/out/cocoa_cb_common.swift

```diff
@@ -516,6 +516,17 @@ class CocoaCB: NSObject {
                 return VO_TRUE
             }
             return VO_FALSE
+        case VOCTRL_GET_UNFS_WINDOW_SIZE:
+            let sizeData = data?.assumingMemoryBound(to: Int32.self)
+            let size = UnsafeMutableBufferPointer(start: sizeData, count: 2)
+            var rect = ccb.window?.unfsContentFrame ?? NSRect(x: 0, y: 0, width: 1280, height: 720)
+            if let screen = ccb.window?.currentScreen, !Bool(ccb.mpv?.opts.hidpi_window_scale ?? 0) {
+                rect = screen.convertRectToBacking(rect)
```

Comment:
about 1, my intention isn't really synchronisation. it's just the macOS quirk that UI updates need to run on the main queue (in the other cases, also async and not sync). in this case we just access our own variable. if run sync it might be possible to run in a deadlock or at least something might get temporarily stuck.